container/go/cmd/pusher/pusher.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// This binary pushes an image to a Docker Registry.
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+
+	"github.com/bazelbuild/rules_docker/container/go/pkg/oci"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/pkg/errors"
+)
+
+var (
+	dst             = flag.String("dst", "", "The destination location including repo and digest/tag of the docker image to push. Supports fully-qualified tag or digest references.")
+	src             = flag.String("src", "", "The path of the source index relative to the execution workspace.")
+	format          = flag.String("format", "", "The form to push, oci or docker (A tarball).")
```

Comment:
It's technically the format of the source image. The format of the pushed image is not really controlled by this binary. It's determined by GCR (I think).

So I would say `The format of the source image, oci|docker. The docker format should be a tarball of the image as generated by docker save.`